cpp/util/etcd.cc

```diff
@@ -688,14 +726,24 @@ void EtcdClient::FetchDone(RequestState* etcd_req, Task* task) {
   VLOG(2) << "EtcdClient::FetchDone: " << task->status();
 
   if (!task->status().ok()) {
-    // TODO(pphaneuf): If there is a connection problem, we should re-do the
-    // request on another node.
+    if (task->status().error_code() == util::error::UNAVAILABLE) {
+      // Seems etcd wasn't available; pick a new etcd server and retry
+      LOG(WARNING) << "Etcd fetch failed: " << task->status() << ", retrying "
+                   << "on next etcd server.";
+      ChooseNextServer();
```

Comment:
Oh, and not only do you need to use `etcds_`, but when you get a redirect, you should add (or move) it to the front of the list.